src/main/java/uk/gov/companieshouse/pscdataapi/transform/CompanyPscTransformer.java

```diff
@@ -49,425 +42,200 @@ public class CompanyPscTransformer {
-                }
-                if (pscDocument.getData().getAddress().getCountry() != null) {
-                    address.setCountry(pscDocument.getData().getAddress().getCountry());
-                }
-                if (pscDocument.getData().getAddress().getLocality() != null) {
-                    address.setLocality(pscDocument.getData().getAddress().getLocality());
-                }
-                if (pscDocument.getData().getAddress().getPostalCode() != null) {
-                    address.setPostalCode(pscDocument.getData().getAddress().getPostalCode());
-                }
-                if (pscDocument.getData().getAddress().getPremises() != null) {
-                    address.setPremises(pscDocument.getData().getAddress().getPremises());
-                }
-                if (pscDocument.getData().getAddress().getRegion() != null) {
-                    address.setRegion(pscDocument.getData().getAddress().getRegion());
-                }
-                if (pscDocument.getData().getAddress().getCareOf() != null) {
-                    address.setCareOf(pscDocument.getData().getAddress().getCareOf());
-                }
-                if (pscDocument.getData().getAddress().getPoBox() != null) {
-                    address.setPoBox(pscDocument.getData().getAddress().getPoBox());
-                }
-                individualBeneficialOwner.setAddress(convertToBoAddress(address));
-            }
-            if (pscDocument.getData().getNaturesOfControl() != null) {
-                individualBeneficialOwner
-                        .setNaturesOfControl(pscDocument.getData().getNaturesOfControl());
-            }
-            if (pscDocument.getData().getLinks() != null) {
-                individualBeneficialOwner.setLinks(pscDocument.getData().getLinks());
-            }
-            if (pscDocument.getData().getNationality() != null) {
-                individualBeneficialOwner.setNationality(pscDocument.getData().getNationality());
-            }
-            if (pscDocument.getData().getSanctioned() != null) {
-                individualBeneficialOwner.setIsSanctioned(pscDocument.getData().getSanctioned());
-            }
-            DateOfBirth getDateOfBirthMapping = mappingDateOfBirth(
-                    optionalPscDocument, registerView);
-            individualBeneficialOwner.setDateOfBirth(getDateOfBirthMapping);
-
-            return individualBeneficialOwner;
-        } else {
-            logger.error("Skipped transforming pscDoc to individualBeneficialOwner");
-            throw new ResourceNotFoundException(HttpStatus.NOT_FOUND,"PscDocument not found");
-        }
-
+            PscDocument pscDocument, Boolean registerView) {
+        logger.info("Attempting to transform pscDocument to IndividualBeneficialOwner",
+                DataMapHolder.getLogMap());
+        IndividualBeneficialOwner individualBo = new IndividualBeneficialOwner();
+        individualBo.setEtag(pscDocument.getData().getEtag());
```

Comment:
I like that we are removing the unneeded null checks. The fields where we call document.getData().getX() need to be inside a check that data is not null though or we can get a null pointer exception